.github/workflows/tests.yml

```diff
@@ -154,6 +145,51 @@ jobs:
           name: test-matrix
           path: semgrep/test-matrix.json
 
+  public-repos-test:
+    name: run semgrep on public repos
+    runs-on: ubuntu-latest
+    needs: [build-core]
+    strategy:
+      matrix:
+        python: [3.7]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Download artifacts
+        uses: actions/download-artifact@v1
+        with:
+          name: ocaml-build-artifacts
+      - name: Install artifacts
+        run: |
+          tar xf ocaml-build-artifacts/ocaml-build-artifacts.tgz
+          sudo cp ocaml-build-artifacts/bin/* /usr/bin
+      - name: Install semgrep
+        run: |
+          cd semgrep
+          export PATH=/github/home/.local/bin:$PATH
+          pip3 install pipenv
+          pipenv install --dev
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/semgrep-cache
+          key: semgrep-repo-cache-test
+      - name: create cache dir if not exists
+        run: mkdir -p ~/.cache/semgrep-cache
+      - run: touch ~/.cache/semgrep-cache/ok
+      - name: Test semgrep
+        run: |
+          cd semgrep
+          export PATH=/github/home/.local/bin:$PATH
+          pipenv run pytest -n auto -v --tb=short --durations=0 tests/qa/run_public_repos.py
```

Comment:
Hmm, I like this change! The less we're relying on magic Makefile commands and non-standard flags the better, in my opinion. However, I think this behavior is a bit non-obvious at first. It took me a minute to realize why `pytest ... tests/` wouldn't also run this test (TL;DR because the filename isn't prefixed with `test_`). Personally, if I run `pytest tests/` I would expect this to run every Python test we have.

Here's my thought: would it make more sense to keep conventional naming here with `test_public_repos.py`? We could then do something like `pytest --ignore=tests/qa/test_public_repos.py ... tests/` above, and `pytest ... tests/qa/test_public_repos.py` here. I think that would be most explicit about what's going on. Thoughts?

Reply:
Yeah that does seem more obvious. I'll make that change before merging